core/src/main/java/com/linecorp/armeria/common/HttpHeaderGetters.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.Streams;
+
+import io.netty.util.AsciiString;
+
+/**
+ * Provides the getter methods to {@link HttpHeaders} and {@link HttpHeadersBuilder}.
+ */
+public interface HttpHeaderGetters extends Iterable<Entry<AsciiString, String>> {
+
+    /**
+     * Tells whether the headers will be the last frame in an HTTP/2 stream.
+     */
+    boolean isEndOfStream();
+
+    /**
+     * Returns the value of the {@code "content-type"} header.
+     *
+     * @return the valid header value if present. {@code null} otherwise.
+     */
+    @Nullable
+    MediaType contentType();
+
+    /**
+     * Returns the value of a header with the specified {@code name}. If there are more than one value for
+     * the specified {@code name}, the first value in insertion order is returned.
+     *
+     * @param name the name of the header to retrieve
+     * @return the first header value if the header is found. {@code null} if there's no such header
+     */
+    @Nullable
+    String get(CharSequence name);
+
+    /**
+     * Returns the value of a header with the specified {@code name}. If there are more than one value for
+     * the specified {@code name}, the first value in insertion order is returned.
+     *
+     * @param name the name of the header to retrieve
+     * @param defaultValue the default value
+     * @return the first header value or {@code defaultValue} if there is no such header
+     */
+    String get(CharSequence name, CharSequence defaultValue);
```

Comment:
Since the return type is `String` I'd expect `defaultValue` to also be `String`.

Reply:
I chose `CharSequence` because the setter methods accept `CharSequence` as a value. Internally, a `CharSequence` value is kept as it is and is converted to a `String` only when required. `defaultValue` is similar - when a header exists, no conversion is made.

After thinking a little bit more, I started to think this kind of optimization might be of dubious value. What do you think? Let me change all value types to `String` if you agree.

Reply:
Sounds good

Reply:
Done.